google/cloud/aiplatform/metadata/execution.py

```diff
@@ -31,6 +34,51 @@ class _Execution(_Resource):
     _resource_noun = "executions"
     _getter_method = "get_execution"
 
+    def add_artifact(
+        self, artifact_resource_name: str, input: bool,
+    ):
+        """Connect Artifact to a given Execution.
+
+        Args:
+            artifact_resource_name (str):
+                Required. The full resource name of the Artifact to connect to the Execution through an Event.
+            input (bool)
+                Required. Whether Artifact is an input event to the Execution or not.
+        """
+
+        event = Event(
+            artifact=artifact_resource_name,
+            type_=Event.Type.INPUT if input else Event.Type.OUTPUT,
+        )
+
+        self.api_client.add_execution_events(
+            execution=self.resource_name, events=[event],
+        )
+
+    def query_input_and_output_artifacts(self) -> Sequence[_Artifact]:
+        """query the input and output artifacts connected to the execution.
+
+        Returns:
+              A Sequence of _Artifacts
+        """
+
+        try:
+            artifacts = self.api_client.query_execution_inputs_and_outputs(
+                execution=self.resource_name
+            ).artifacts
+        except exceptions.NotFound:
```

Comment:
If the execution does not have any artifacts associated with it then I think it would have a NotFound error. For our current use case if the run is setup correctly, then this should not happen, but this could be an issue if we use this API for other use cases in the future.